setLocation.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"errors"
+	"encoding/json"
+	"fmt"
+	"github.com/go-telegram-bot-api/telegram-bot-api"
+	"log"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+)
+
+// Return from www.cepaberto.com
+type cepResponse struct {
+	CEP        string  `json:"cep"`
+	Logradouro string  `json:"logradouro"`
+	Bairro     string  `json:"bairro"`
+	Cidade     string  `json:"cidade"`
+	Estado     string  `json:"estado"`
+	Latitude   string  `json:"latitude"`
+	Longitude  string  `json:"longitude"`
+	Altitude   float64 `json:"altitude"`
+	DDD        int     `json:"ddd"`
+	IBGE       string  `json:"ibge"`
+}
+
+// locationHandler receive postal code from user.
+func locationHandler(update tgbotapi.Update, bot *tgbotapi.BotAPI, config botConfig, msgs botMessages) error {
+
+	tmp := strings.ToLower(strings.Trim(update.Message.CommandArguments(), " "))
+	tmp = strings.Replace(tmp, "-", "", -1)
+	tmp = strings.Replace(tmp, ".", "", -1)
+	tmp = strings.Replace(tmp, "/", "", -1)
+	args := strings.Split(tmp, " ")
+	user := update.Message.From
+	cep := ""
+
+	cmd := strings.ToLower(update.Message.Command())
+
+	if cmd == "setlocation" && len(args) != 2 {
```

Comment:
I'd do this block of code differently. Something like:

```golang
cep := ""

switch cmd {
case "setlocation":
  if len(args) != 2 {
    return errors.New("/setlocation <pais> <código postal>")
  }
  country := args[0]
  if country != "br" {
    return fmt.Errorf("País invalido: %q", country)
  }
  cep = args[1]
case "cep":
  if len(args) != 1 {
    return errors.New("CEP não especificado.")
  }
  cep = args[0]
}

if err := findCEP(user,cep,config); err != nil {
  return errors.New("Não foi possível achar a sua localização . CEP "+cep);
}
sendReply(... mgs.LocationSuccess)
return nil
```

Note:

- In go we almost never use else. Instead of:

```golang
if (foo) {
  a = 1
} else {
  a = 2
}
```
We normally do:

```golang
a = 2
if (foo) {
  a = 1
}
```
Also, only use an error to represent an error (not to return a regular message). The reason is that in the future we may process errors differently in the caller. Use a sendReply to send a regular message, and only fail with err != nil in case of real errors.